### DISCUSSION TITLE
Option to use multiprocessing lib instead of threading

### DISCUSSION THREAD

**Discussion by coder2020official** [eternnoir/pyTelegramBotAPI#1923] [Ideas]
Hi.
I am working on it, what do you think if there would be such option?

**Comment by coder2020official** [comment]
@Badiboy

**Comment by Badiboy** [comment]
Multiprocessing is good for scaling, but it does not allow (in general) to have shared data between processes, except some special data pipes. Thus multiprocessing version will have significant limitations for application.

Additionally, AFAIK, it is adviced to use WSGI for multiprocessing instead of embedding it in the bot. But I have no competences in WSGI.
